app/src/main/java/com/kamron/pogoiv/logic/PokeInfoCalculator.java

```diff
@@ -190,32 +190,25 @@ public IVScanResult getIVPossibilities(Pokemon selectedPokemon, double estimated
         //IV vars for lower and upper end cp ranges
 
 
-        IVScanResult returner;
-        //It's safe to proceed if *one* is not 10, though it takes a bit longer.
-        if (pokemonHP != 10 || pokemonCP != 10) {
-            returner = ScanContainer.createIVScanResult(selectedPokemon, estimatedPokemonLevel, pokemonCP, false);
-            for (int staminaIV = 0; staminaIV < 16; staminaIV++) {
-                int hp = (int) Math.max(Math.floor((baseStamina + staminaIV) * lvlScalar), 10);
-                if (hp == pokemonHP) {
-                    double lvlScalarStamina = Math.sqrt(baseStamina + staminaIV) * lvlScalarPow2;
-                    //Possible STA IV
-                    //System.out.println("Checking sta: " + staminaIV + ", gives " + hp);
-                    for (int defenseIV = 0; defenseIV < 16; defenseIV++) {
-                        for (int attackIV = 0; attackIV < 16; attackIV++) {
-                            int cp = (int) Math.floor(
-                                    (baseAttack + attackIV) * Math.sqrt(baseDefense + defenseIV) * lvlScalarStamina);
-                            if (cp == pokemonCP) {
-                                returner.addIVCombination(attackIV, defenseIV, staminaIV);
-                            }
+        IVScanResult returner = ScanContainer.createIVScanResult(selectedPokemon, estimatedPokemonLevel, pokemonCP);
+        for (int staminaIV = 0; staminaIV < 16; staminaIV++) {
+            int hp = (int) Math.max(Math.floor((baseStamina + staminaIV) * lvlScalar), 10);
+            if (hp == pokemonHP) {
+                double lvlScalarStamina = Math.sqrt(baseStamina + staminaIV) * lvlScalarPow2;
+                for (int defenseIV = 0; defenseIV < 16; defenseIV++) {
+                    for (int attackIV = 0; attackIV < 16; attackIV++) {
+                        int cp = (int) Math.floor(
+                                (baseAttack + attackIV) * Math.sqrt(baseDefense + defenseIV) * lvlScalarStamina);
```

Comment:
@NightMadness So if I understand you correctly, instead of `int cp = (int) Math.floor` we need to do like for `hp` above, right?
```java
int cp = (int) Math.max(10, Math.floor(...));
```